kura/org.eclipse.kura.api/META-INF/MANIFEST.MF

```diff
@@ -29,13 +29,13 @@ Export-Package: org.eclipse.kura;version="1.3.0",
  org.eclipse.kura.gpio;version="1.0.1",
  org.eclipse.kura.linux.udev;version="1.0.0",
  org.eclipse.kura.message;version="1.1.0",
- org.eclipse.kura.net;version="1.2.1",
+ org.eclipse.kura.net;version="1.2.2",
```

Comment:
First, I think a minor version increase is okay here. We try to follow the OSGi semantic versioning model. In this case, we are introducing a breaking change for the provider, not the consumer. The consumer is free to continue using the deprecated method, we are just indicating there is a better option. The consumer should be importing [1.2, 2), so they should be okay. I doubt there are many users of Kura re-implementing this interface. 

Second, while I agree with @kartben, didn't we decide that Kura v3.0.0 was not going to be backward compatible to v2.x? So, if we are going to make breaking changes to the API, now would be the time. As mentioned above, I don't think this is the case here, but just a general thought on the next release.